dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java

```diff
@@ -452,7 +452,16 @@ public boolean equals(Object o) {
             && timeoutFlag == that.timeoutFlag
             && timeoutNotifyStrategy == that.timeoutNotifyStrategy
             && Objects.equals(resourceIds, that.resourceIds)
-            && environmentCode == that.environmentCode;
+            && environmentCode == that.environmentCode
+            && version == that.version
+            && projectCode == that.projectCode
```

Comment:
The `equals ` method cannot be modified here. This method is to prevent invalid version growth